consensus/spos/bls/subroundEndRound.go

```diff
@@ -250,6 +252,17 @@ func (sr *subroundEndRound) doEndRoundJobByParticipant(cnsDta *consensus.Message
 
 	sr.SetProcessingBlock(true)
 
+	shouldNotCommitBlock := sr.ExtendedCalled || int64(header.GetRound()) < sr.Rounder().Index()
+	if shouldNotCommitBlock {
```

Comment:
Yes, there was an edge case when extended could be called because the round time is ended, very close before to the call of this method, and if node would have been a proposer next round it would started to create the new block while here it still would have been in commit block